bundles/EcommerceFrameworkBundle/IndexService/ProductList/ElasticSearch/AbstractElasticSearch.php

```diff
@@ -1135,6 +1159,65 @@ protected function doLoadGroupByValues()
         $this->preparedGroupByValuesLoaded = true;
     }
 
+    /**
+     * Deep search for buckets in result aggregations array, as the structure of the result array
+     * may differ dependent on the used aggregations (i.e. date filters, nested aggr, ...)
+     *
+     * @param array $aggregations
+     * @return array
+     */
+    private function searchForBuckets(array $aggregations)
+    {
+        if (array_key_exists("buckets", $aggregations)) {
+            return $aggregations["buckets"];
+        }
+
+        // usually the relevant key is at the very end of the array so we reverse the order
+        $aggregations = array_reverse($aggregations, true);
+
+        foreach ($aggregations as $aggregation) {
+            if (!is_array($aggregation)) {
+                continue;
+            }
+            $buckets = $this->searchForBuckets($aggregation);
+            if (!empty($buckets)) {
+                return $buckets;
+            }
+        }
+
+        return [];
+    }
+
+    /**
+     * Recursively convert aggregation data (sub-aggregations possible)
+     *
+     * @param array $bucket
+     * @return array
+     */
+    private function convertBucketValues(array $bucket){
```

Comment:
shouldn't we make these methods protected to make overwriting possible?

Reply:
makes sense - resolved